rclcpp/test/rclcpp/test_node.cpp

```diff
@@ -2455,3 +2490,89 @@ TEST_F(TestNode, get_publishers_subscriptions_info_by_topic) {
     subscription_list = node->get_subscriptions_info_by_topic("13");
   }, rclcpp::exceptions::InvalidTopicNameError);
 }
+
+TEST_F(TestNode, callback_groups) {
+  auto node = std::make_shared<rclcpp::Node>("node", "ns");
+  size_t num_callback_groups_in_basic_node = node->get_callback_groups().size();
+
+  auto group1 = node->create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
+  EXPECT_EQ(1u + num_callback_groups_in_basic_node, node->get_callback_groups().size());
+
+  auto group2 = node->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
+  EXPECT_EQ(2u + num_callback_groups_in_basic_node, node->get_callback_groups().size());
+}
+
+// This is tested more thoroughly in node_interfaces/test_node_graph
+TEST_F(TestNode, get_entity_names) {
+  auto node = std::make_shared<rclcpp::Node>("node", "ns");
+  const auto node_names = node->get_node_names();
+  EXPECT_NE(
+    node_names.end(),
+    std::find(node_names.begin(), node_names.end(), node->get_fully_qualified_name()));
```

Comment:
Would it make sense to go for a positive test here?  That is, something like:

```
EXPECT_EQ("/ns/node", std::find(node_names.begin(), node_names.end(), node->get_fully_qualified_name()));
```

(as a side note, reviewing this is exactly why I don't like `auto`.  In order for me to tell what type `node->get_node_names()` returned, I had to go looking at the header file, which isn't part of this PR.  It's not the end of the world, but just makes life more difficult for the reader)

Reply:
I'm not sure it adds anything. Checking that find didn't return `end()` would still be required in some form because dereferencing `end()` would crash the test. Likewise, we're not checking that `std::find` works, just that the node's name is in the collection of node_names.

My counter-argument would be that this unit test is robust to changes in the return type collection (as long as that type is suitable for std::find), which is also valuable.

Reply:
> I'm not sure it adds anything. Checking that find didn't return `end()` would still be required in some form because dereferencing `end()` would crash the test. Likewise, we're not checking that `std::find` works, just that the node's name is in the collection of node_names.

That's a good point.  OK, I'm fine with this test as-is.